src/api/component.js

```diff
@@ -14,8 +14,10 @@ import getAllKeys from '../util/get-all-keys';
 import getOwnPropertyDescriptors from '../util/get-own-property-descriptors';
 import getSetProps from './props';
 import syncPropToAttr from '../util/sync-prop-to-attr';
+import root from 'window-or-global';
 
-const { HTMLElement } = window;
+const { HTMLElement } = root;
+const htmlElementPrototype = HTMLElement ? HTMLElement.prototype : {};
```

Comment:
If `HTMLElement` isn't available, this would cause errors below. Instead, we just create a prototype from the base `Object` even if it's not used.